editor/src/components/canvas/ui/floating-insert-menu-helpers.ts

```diff
@@ -0,0 +1,15 @@
+export type InsertMenuMode = 'closed' | 'insert' | 'convert' | 'wrap'
+
+export const insertMenuModes: {
```

Comment:
I suspect this approach might be overkill, but if that's the case it wouldn't take much to change the code later anyway